Cannot find module '.' or its corresponding  type declarations

<!-- Thanks for your interest in cheerio!

Please note that issues should be primarily used for tracking bugs and feature requests.
If you have a more general question, please consider consulting StackOverflow first:
https://stackoverflow.com/questions/tagged/cheerio

If you think you uncovered a bug, please try to provide a minimal example that triggers the behavior.
Please note that we will not investigate issues that perform HTTP requests, as the source might already have changed.
-->
when I use cheerio in my project with typescript,  and try to compile it by `tsc`. The compiler throws an exception described below:
```typescript
error TS2307: Cannot find module '.' or its corresponding type declarations.

2 import type { CheerioAPI, Cheerio } from '.'; 

Found 1 error in node_modules/cheerio/lib/esm/static.d.ts:2
```
This seems to be a bug in the package itself. How should I solve this problem? Thanks.

And I only use cheerio in my project like this:
```typescript
import * as cheerio from "cheerio";
cheerio.load(content) // The type of content is string.
```
"cheerio": "^1.0.0-rc.12"
"@types/cheerio": "^0.22.31"
node v16.16.0
npm v8.16.0

--- a/src/__tests__/deprecated.spec.ts
+++ b/src/__tests__/deprecated.spec.ts
@@ -4,7 +4,7 @@
  * maintained until that time.
  */
 import * as fixtures from '../__fixtures__/fixtures.js';
-import cheerio from '..';
+import cheerio from '../index.js';
 
 describe('deprecated APIs', () => {
   /**
@@ -244,7 +244,8 @@ describe('deprecated APIs', () => {
      * ```
      *
      * @example <caption>To render the markup of an entire document, invoke the
-     * `html` function exported by the Cheerio module with a "root" selection.</caption>
+     * `html` function exported by the Cheerio module with a "root"
+     * selection.</caption>
      *
      * ```js
      * cheerio.html($.root());

--- a/src/__tests__/xml.spec.ts
+++ b/src/__tests__/xml.spec.ts
@@ -1,4 +1,4 @@
-import cheerio from '..';
+import cheerio from '../index.js';
 import type { CheerioOptions } from '../options.js';
 
 function xml(str: string, options?: CheerioOptions) {

--- a/src/api/attributes.spec.ts
+++ b/src/api/attributes.spec.ts
@@ -1,4 +1,4 @@
-import cheerio from '..';
+import cheerio from '../index.js';
 import type { Cheerio } from '../cheerio.js';
 import type { Element } from 'domhandler';
 import {

--- a/src/api/css.spec.ts
+++ b/src/api/css.spec.ts
@@ -1,4 +1,4 @@
-import cheerio from '..';
+import cheerio from '../index.js';
 import type { Cheerio } from '../cheerio.js';
 import type { Element } from 'domhandler';
 import { mixedText } from '../__fixtures__/fixtures.js';

--- a/src/api/manipulation.spec.ts
+++ b/src/api/manipulation.spec.ts
@@ -1,5 +1,5 @@
 import { load } from '../../src';
-import type { CheerioAPI, Cheerio } from '..';
+import type { CheerioAPI, Cheerio } from '../index.js';
 import { fruits, divcontainers, mixedText } from '../__fixtures__/fixtures.js';
 import type { AnyNode, Element } from 'domhandler';
 

--- a/src/cheerio.spec.ts
+++ b/src/cheerio.spec.ts
@@ -1,12 +1,12 @@
 import { parseDOM } from 'htmlparser2';
-import cheerio from '.';
+import cheerio from './index.js';
 import * as utils from './utils.js';
 import { fruits, food, noscript } from './__fixtures__/fixtures.js';
 import type { Cheerio } from './cheerio.js';
 import type { Element } from 'domhandler';
 import type { CheerioOptions } from './options.js';
 
-declare module '.' {
+declare module './index.js' {
   interface Cheerio<T> {
     myPlugin(...args: unknown[]): {
       context: Cheerio<T>;

--- a/src/load.spec.ts
+++ b/src/load.spec.ts
@@ -1,4 +1,4 @@
-import cheerio from '.';
+import cheerio from './index.js';
 
 describe('.load', () => {
   it('(html) : should retain original root after creating a new node', () => {

--- a/src/slim.ts
+++ b/src/slim.ts
@@ -10,7 +10,7 @@ export type {
   ParentNode,
   Element,
   Document,
-} from '.';
+} from './index.js';
 
 /**
  * Types used in signatures of Cheerio methods.
@@ -25,11 +25,13 @@ import render from 'dom-serializer';
 import { parseDocument } from 'htmlparser2';
 
 /**
- * Create a querying function, bound to a document created from the provided markup.
+ * Create a querying function, bound to a document created from the provided
+ * markup.
  *
  * @param content - Markup to be loaded.
  * @param options - Options for the created instance.
- * @param isDocument - Always `false` here, as we are always using `htmlparser2`.
+ * @param isDocument - Always `false` here, as we are always using
+ *   `htmlparser2`.
  * @returns The loaded document.
  * @see {@link https://cheerio.js.org#loading} for additional usage information.
  */

--- a/src/static.spec.ts
+++ b/src/static.spec.ts
@@ -1,5 +1,5 @@
 import * as fixtures from './__fixtures__/fixtures.js';
-import cheerio, { CheerioAPI } from '.';
+import cheerio, { CheerioAPI } from './index.js';
 
 describe('cheerio', () => {
   describe('.html', () => {

--- a/src/static.ts
+++ b/src/static.ts
@@ -1,5 +1,5 @@
 import type { BasicAcceptedElems } from './types.js';
-import type { CheerioAPI, Cheerio } from '.';
+import type { CheerioAPI, Cheerio } from './index.js';
 import type { AnyNode, Document } from 'domhandler';
 import { textContent } from 'domutils';
 import {

